### PR TITLE
chore: rename RETURN to RETURNS in masking policy rule; check masking policy rule input type and return type

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. rename RETURN to RETURNS in masking policy rule; 
2. check masking policy rule input type and return type, add test case.
3. refactor error code

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12406)

### CODE REVIEW

File: src/meta/app/src/data_mask/mod.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ impl Display for DatamaskId {
 
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
 pub struct DatamaskMeta {
+    // Vec<(arg_name, arg_type)>
```
Comment (BohuTANG) [inline src/meta/app/src/data_mask/mod.rs:51]: Remove this line?

File: src/query/sql/src/planner/binder/ddl/data_mask.rs [hunk 2]
```diff
@@ -33,6 +34,16 @@ impl Binder {
             policy,
         } = stmt;
 
+        // check if input type match to the return type
+        let return_type = policy.return_type.to_string().to_lowercase();
```
Comment (BohuTANG) [inline src/query/sql/src/planner/binder/ddl/data_mask.rs:38]: Need a logic test for this

Comment (lichuang) [reply src/query/sql/src/planner/binder/ddl/data_mask.rs:38]: see: tests/sqllogictests/suites/ee/05_ee_ddl/05_0001_ddl_data_mask in this pr

**Review by BohuTANG — APPROVED** [review]

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 13 of 17 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12406)* status: 14 of 17 files reviewed, 3 unresolved discussions (waiting on @BohuTANG and @lichuang)

___
*[`src/meta/app/src/app_error.rs` line 389 at r2](https://reviewable.io/reviews/datafuselabs/databend/12406#-NbSPsfk2zhLra7ehZ7V:-NbSPsfk2zhLra7ehZ7W:b-darsj1) ([raw file](https://github.com/datafuselabs/databend/blob/f0eba5f1b9875c382df159c1ca0c7b8c01b083c2/src/meta/app/src/app_error.rs#L389)):*
> ```XML
>         arg_type: impl Into<String>,
>         return_type: impl Into<String>,
>         context: impl Into<String>,
> ```

`impl ToString` is usually used in such scenario.



_Code quote:_
```XML
        arg_name: impl Into<String>,
        arg_type: impl Into<String>,
        return_type: impl Into<String>,
        context: impl Into<String>,
```

**Review by drmingdrmer — APPROVED** [review]

**Pull request merged**